lib/ddtrace/configuration/settings.rb

```diff
@@ -198,6 +314,9 @@ def runtime_metrics(options = nil)
         settings
       end
 
+      # TODO: remove me. Use `c.runtime_metrics.enabled =`
+      # @deprecated Use `runtime_metrics.enabled` instead.
+      # @return [Boolean]
       option :runtime_metrics_enabled do |o|
```

Comment:
Minor: May be worth stating here that this is deprecated for removal (see also [how Java does it nowadays](https://docs.oracle.com/javase/9/core/enhanced-deprecation1.htm#JSCOR-GUID-E182DAF6-282B-4C1F-8CCF-0A61008E3F11))

Reply:
I'm thinking that something marked as deprecated can be removed in the next major version unceremoniously.

What do you think?

(I do like how explicit `@Deprecated(since="9", forRemoval=true)` reads!)

Reply:
> I'm thinking that something marked as deprecated can be removed in the next major version unceremoniously.

I'm happy with that definition, but my concern is that it may not be clear to customers.

It's a bit like that for Java as well -- some APIs have been deprecated for 10+ (20+?) years and they're still there. So I think it's important to clarify that we mean the "it's deprecated because we're going to remove it soon" and not the "it's deprecated because you shouldn't use it, but well, we're going to keep it here".

Perhaps we could add a small text block explaining what deprecation means in our library and link it whenever we mark a method as deprecated? Something like

```
# @deprecated [For removal](...nice link so we don't repeat ourselves...), use `runtime_metrics.enabled` instead.
```

Reply:
Done this suggestion: 
![Screen Shot 2021-12-21 at 2 13 19 PM](https://user-images.githubusercontent.com/583503/147004464-1087977b-fc58-4c52-a897-820590726fd3.png)

Which links to: 
![Screen Shot 2021-12-21 at 2 13 23 PM](https://user-images.githubusercontent.com/583503/147004475-1a534ca2-3c2b-44e8-ac12-e6dd4ded641c.png)

Reply:
Seems great, thanks!